src/diffpy/utils/diffraction_objects.py

```diff
@@ -255,7 +256,9 @@ def _check_operation_compatibility(self, other):
             raise TypeError(invalid_add_type_emsg)
         if isinstance(other, DiffractionObject):
             if self.all_arrays.shape != other.all_arrays.shape:
```

Comment:
We need to first check the shape, otherwise we get ` ValueError: operands could not be broadcast together with shapes (0,3) (2,3)` error message.